darts/timeseries.py

```diff
@@ -911,6 +935,160 @@ def univariate_values(self, copy=True, sample=0) -> np.ndarray:
+            ignore_time_axes : bool, default False
+                Ignore errors when time axis is varies for some time series. Note that this may yield unexpected results
+
+            Returns
+            -------
+            TimeSeries
+                concatenated timeseries
+        """
+
+        axis = TimeSeries._get_str_axis(axis)
+        if len(timeserie_sequence) < 2:
+            return timeserie_sequence
+
+        start_date_equal = len(set([ts.start_time() for ts in timeserie_sequence])) == 1
+        end_date_equal = len(set([ts.end_time() for ts in timeserie_sequence])) == 1
+        freq_equal = len(set([ts.freq_str for ts in timeserie_sequence])) == 1
+        time_axis_equal = start_date_equal and end_date_equal and freq_equal
+        component_axis_equal = len(set([ts.width for ts in timeserie_sequence])) == 1
+        sample_axis_equal = len(set([ts.n_samples for ts in timeserie_sequence])) == 1
+
+        if ((axis == 'time' and not (component_axis_equal and sample_axis_equal)) or
+            (axis == 'component' and not ((time_axis_equal or ignore_time_axes) and sample_axis_equal)) or
+            (axis == 'sample' and not ((time_axis_equal or ignore_time_axes) and component_axis_equal))):
+
+            raise_log(AttributeError('Remaining (non-concatenating) axes need to be equal.'))
+
+        da_sequence = [ts.data_array() for ts in timeserie_sequence]
+
+        if axis == DIMS[0]: # time
+            if time_axis_equal:
+                if ignore_time_axes:
+                    da_concat=da_sequence[0]
+                else:
+                    raise_log(AttributeError("All concatenated have the same time axis. "
+                                             "If this is not an error, use `ignore_time_axis=True`."))
+            else:
+                consecutive_time_axes = True
+                for i in range(1, len(timeserie_sequence)):
+                    if timeserie_sequence[i - 1].end_time() + pd.Timedelta(timeserie_sequence[0].freq) != \
+                            timeserie_sequence[i].start_time():
+                        consecutive_time_axes = False
+                        break
+
+                if consecutive_time_axes:
+                    da_concat = xr.concat(da_sequence, dim=axis)
+                else:
+                    raise_log(AttributeError("When concatenating over time axis, all series need to be subsequent, i.e."
+                                             " end_time of the previous one should be equal to start_time - 1 of the "
+                                             "following timeserie."
+                                         ))
+
+        else: # component or sample
+            if len(set([ts.shape[0] for ts in da_sequence])) != 1:
+                raise AttributeError("All concatenating time series need to have the same time axis or at least"
+                                     " be of the same length.")
```

Comment:
I think to avoid confusion we should either have
* `time_axis_equal == True`, or
* `ignore_time_axes == True` and `len(set([ts.shape[0] for ts in da_sequence])) == 1`.

I wouldn't silently concatenate series that do not have the same time axes even if they have the same length. This is basically what is implemented here, except that the check `len(set([ts.shape[0] for ts in da_sequence])) == 1` could be move and not be done preemptively, but performed only if `time_axis_equal == False`.